### PR TITLE
Users/shilpir/sqlfix:Changes to make "Specify Firewall Rules using" input values dynamic

### PR DESCRIPTION
**Pull request by ShilpiR**
1. Changes to make the picklist values for  "Specify Firewall Rules using" input dynamic based on whether the Service is on cloud or on-premise.

2. Corresponding changes in task.json and powershell script

2. Removed custom code for masking password.

### CODE REVIEW

File: Tasks/SqlAzureDacpacDeployment/task.json [hunk 1]
```diff
@@ -151,8 +147,15 @@
 			"defaultValue": "true",
 			"helpMarkDown": "If selected, deletes the IP Address exception in the Azure SQL DB Firewall after the task ends"
 		}
-	],
-	"sourceDefinitions": [],
+	],	
+    "sourceDefinitions": [ 
+     { 
+         "target": "IpDetectionMethod",
```
Comment (pankajkhanzode) [inline Tasks/SqlAzureDacpacDeployment/task.json]: This is good change. It is good that service ie telling what it supports. Right way to model client driven by service contract.

**Comment by pankajkhanzode** [comment]
Signed off.

File: Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1 [hunk 2]
```diff
@@ -41,7 +41,7 @@ $sqlPackageOnTargetMachineBlock = Get-Content $sqlDeploymentScriptPath | Out-Str
 
 $sqlPackageArguments = Get-SqlPackageCommandArguments -dacpacFile $dacpacFile -targetMethod $targetMethod -serverName $serverName -databaseName $databaseName -sqlUsername $sqlUsername -sqlPassword $sqlPassword -connectionString $connectionString -publishProfile $publishProfile -additionalArguments $additionalArguments
 
-$scriptArguments = "-sqlPackageArguments $sqlPackageArguments -sqlPassword $sqlPassword -targetConnectionString $connectionString"
```
Comment (kmkumaran) [inline Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1]: Can you help me why this change?

Comment (ShilpiR) [inline Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1]: SqlPassword and TargetConnectionString are needed here to mask the password. Since we decided that we will rely on Secure Variables and not do any custom processing, I am not sending the password or connection string now
